packages/react-components/src/components/chart/hooks/useDataZoom.ts

```diff
@@ -108,24 +149,36 @@ export const useDataZoom = (chartRef: MutableRefObject<EChartsType | null>, view
     };
   }, [chartRef, mode, dispatch, setViewport]);
 
-  // Animate viewport changes based on the tick interval
+  /**
+   * Handle setting the dataZoom for the chart
+   * based on the tick interval
+   * convertedViewport is updated in the setInterval loop above
+   * as part of the tick action
+   */
   useEffect(() => {
     const { isDurationViewport, initial, end } = convertedViewport;
-    const chart = chartRef.current;
-    /**
-     * only update the zoom imperatively if we are in live mode
-     * or the date picker picks a new viewport
-     *
-     * It is possible that we have a duration viewport + echarts-gesture combo
-     * so we are explicitly denying echarts-gesture
-     */
-    const shouldUpdate = chart != null || isDurationViewport || lastUpdatedBy === 'date-picker';
-    if (!shouldUpdate || lastUpdatedBy === 'echarts-gesture') return;
 
-    chart?.setOption({
-      dataZoom: { ...DEFAULT_DATA_ZOOM, startValue: initial, endValue: end },
-    });
-  }, [chartRef, convertedViewport, lastUpdatedBy]);
+    if (!isDurationViewport) return;
+
+    zoomChart({ startValue: initial, endValue: end });
+  }, [zoomChart, convertedViewport]);
+
+  /**
+   * any change to the viewport that is not an echarts gesture is synced
+   * to the chart dataZoom
+   */
+  useEffect(() => {
+    const { unsubscribe } = viewportManager.subscribe(group, handleViewportUpdate);
+    return unsubscribe;
+  }, [handleViewportUpdate, group]);
+
+  /**
+   * ensure the chart is zoomed to the current viewport before it
+   * can start reacting to events
+   */
+  useEffectOnce(() => {
+    handleViewportUpdate(viewport ?? DEFAULT_VIEWPORT);
+  });
```

Comment:
Nice! I think using this will prevent bugs from an accidental update of a dependency array, especially when ESLint starts yelling.